packages/entity-server/src/routes/hierarchy/types.ts

```diff
@@ -29,31 +48,36 @@ export type ExtendedFieldFunctions = Readonly<
 
 type ExcludeCommonFields<T, U> = Omit<T, Extract<keyof T, keyof U>>;
 
+export type FlattableEntityFields = Pick<EntityFields, SimpleFieldKeys<EntityFields>>;
+
 export type ExtendedEntityFields = ExcludeCommonFields<EntityFields, ExtendedFieldFunctions> &
   ExtendedFieldFunctions;
 
 export type EntityResponseObject = {
   [field in keyof ExtendedEntityFields]?: ExtendedEntityFields[field];
 };
 
+export type EntityResponse =
+  | EntityResponseObject
+  | FlattableEntityFields[keyof FlattableEntityFields];
```

Comment:
`FlattableEntityFields[keyof FlattableEntityFields]` appears several times in this PR, so it may be worth defining it as a new type (eg `EntityResponseFlat`, `EntityResponsePrimitive`)

Reply:
I did this in pt4 👍